src/mongo/nodes.ts

```diff
@@ -74,13 +54,18 @@ export class MongoServerNode implements IMongoServer {
 		return MongoServerNode.getMongoDatabaseNodes(this._connectionString, this);
 	}
 
-	static async getMongoDatabaseNodes(connectionString: string, parentNode: IMongoServer): Promise<MongoDatabaseNode[]> {
-		const db = await MongoClient.connect(connectionString);
+	static async getMongoDatabaseNodes(connectionString: string, parentNode: IMongoServer): Promise<INode[]> {
+		let db: Db;
 		try {
+			db = await MongoClient.connect(connectionString);
```

Comment:
It looks like you moved this line into the try. Was this line causing an issue? 
I'm trying to understand which of the 3 lines in the try block is expected to raise an error. 
And does error.name have any useful information (like a URIError, ReferenceError, etc.) ?

Reply:
The catch block is most relevant to the first line and the finally block is most relevant to the second and third line.

In other words, I moved the 'MongoClient.connect' line into the try block because it is the line that will throw an error if we can't connect. The other two lines _shouldn't_ throw errors, but I want to be sure `db` is closed even if an error does occur (hence the finally block)